frontend/src/components/misc/tabs/Tabs.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+import styles from './Tabs.module.scss';
+
+export interface Tab {
+    key: string;
+    title: React.ReactNode;
+    content: React.ReactNode;
+    disabled?: boolean;
+}
+
+interface TabsProps {
+    tabs: Array<Tab>;
```

Comment:
Just personal preference since I find this more readable oftentimes. But I'm not bound to it, so if you prefer your suggested notation, I could change that.